ga4gh/cli.py

```diff
@@ -223,23 +261,77 @@ class ReadsSearchRunner(AbstractSearchRunner):
     """
     Runner class for the reads/search method
     """
+    class GASearchReadsRequestGoogle(protocol.ProtocolElement):
+
+        __slots__ = ['end', 'pageSize', 'pageToken', 'readGroupIds',
+                     'referenceName', 'start']
+
+        def __init__(self):
+            self.end = None
+            self.pageSize = None
+            self.pageToken = None
+            self.readGroupIds = []
+            self.referenceName = None
+            self.start = 0
+
     def __init__(self, args):
         super(ReadsSearchRunner, self).__init__(args)
-        request = protocol.GASearchReadsRequest()
+        if self.usingWorkaroundsFor(client.HTTPClient.workaroundGoogle):
+            # google says referenceId not a valid field
+            request = self.GASearchReadsRequestGoogle()
+        else:
+            request = protocol.GASearchReadsRequest()
         setCommaSeparatedAttribute(request, args, 'readGroupIds')
         request.start = args.start
         request.end = args.end
         request.referenceId = args.referenceId
         request.referenceName = args.referenceName
-        if self.usingWorkaroundsFor(client.HTTPClient.workaroundGoogle):
-            # google says referenceId not a valid field
-            del request.__dict__['referenceId']
         self.setHttpClient(request, args)
 
     def run(self):
         self._run('searchReads', 'id')
 
 
+class ReferenceBasesSearchRunner(AbstractSearchRunner):
+    """
+    Runner class for the references/{id}/bases method
+    """
+    def __init__(self, args):
+        super(ReferenceBasesSearchRunner, self).__init__(args)
+        request = protocol.GAListReferenceBasesRequest()
+        request.start = args.start
+        request.end = args.end
+        self._id = args.id
+        self.setHttpClient(request, args)
+
+    def run(self):
+        method = self._httpClient.listReferenceBases
+        for base in method(self._request, self._id):
+            print(base.sequence)
+
+
+class ReferenceSetFindRunner(AbstractFindRunner):
+    """
+    Runner class for the referencesets/{id} method
+    """
+    def __init__(self, args):
+        super(ReferenceSetFindRunner, self).__init__(args)
+
+    def run(self):
```

Comment:
Do we need this indirection? Why not just have
```python
def run(self):
      self._httpClient.getReferenceSet(self._id)
```
If we do need to use the indirection, I'd rather use
```python
def run(self):
      self._run(self._httpClient.getReferenceSet)
```
and save on the strings and getattr calls.